### PR TITLE
[AVAX] Merge Cortina 9 (v1.10.9)

### PR DESCRIPTION
**Pull request by evlekht**
## Why this should be merged
This PR contains changes from ava-labs/avalanchego between cortina-8 and cortina-9.
https://github.com/ava-labs/avalanchego/releases/tag/v1.10.9
### Conflicts
- .github/workflows/buf-push.yml
- .github/workflows/build-deb-pkg.sh
- .github/workflows/build-tgz-pkg.sh
- .github/workflows/publish_image.sh
- .github/workflows/run_e2e_tests.sh
- Dockerfile
- RELEASES.md
- database/mock_iterator.go
- go.mod
- go.sum
- network/certs_test.go
- network/peer/mock_gossip_tracker.go
- network/peer/peer_test.go
- network/peer/upgrader.go
- node/node.go
- scripts/build_avalanche.sh
- scripts/build_image.sh
- scripts/build_local_image.sh
- scripts/constants.sh
- scripts/tests.e2e.sh
- scripts/tests.upgrade.sh
- snow/engine/common/mock_sender.go
- tests/e2e/README.md
- tests/e2e/e2e.go
- tests/e2e/e2e_test.go
- tests/e2e/ping/suites.go
- tests/fixture/testnet/config.go
- version/constants.go
- vms/platformvm/fx/mock_fx.go
- vms/proposervm/batched_vm_test.go
- vms/proposervm/block/build.go
- vms/proposervm/vm_test.go
### Post-merge fixes
See https://github.com/chain4travel/caminogo/pull/308#pullrequestreview-1910750971 .
## How this was tested
With already existing unit-tests and e2e tests.

### CODE REVIEW

File: scripts/constants.sh [hunk 1]
```diff
@@ -3,8 +3,7 @@
 # Use lower_case variables in the scripts and UPPER_CASE variables for override
 # Use the constants.sh for env overrides
 
-# Set the PATHS
-GOPATH="$(go env GOPATH)"
+CAMINOGO_PATH=$( cd "$( dirname "${BASH_SOURCE[0]}" )"; cd .. && pwd )
```
Comment (evlekht) [inline scripts/constants.sh:6]: This `GOPATH` var is not used anywhere, while `CAMINOGO_PATH` is required in this script. 
`constants.sh` is sourced into all other scripts, so usually this var is pre-set. But avax has this line here and some of their scripts not presetting this var, so I decided to include it here as well - the less the diff, the better.

File: scripts/mock.gen.sh [hunk 2]
```diff
@@ -20,7 +20,6 @@ while IFS= read -r line
 do
   IFS='=' read src_import_path interface_name output_path <<< "${line}"
   package_name=$(basename $(dirname $output_path))
-  [[ $src_import_path == \#* ]] && continue
```
Comment (evlekht) [inline scripts/mock.gen.sh:23]: Its removed in avax. Change is lost during previous merges, probably.

File: scripts/mocks.mockgen.txt [hunk 3]
```diff
@@ -56,3 +48,7 @@ github.com/ava-labs/avalanchego/vms/registry=VMRegistry=vms/registry/mock_vm_reg
 github.com/ava-labs/avalanchego/vms=Factory,Manager=vms/mock_manager.go
 github.com/ava-labs/avalanchego/x/merkledb=MerkleDB=x/merkledb/mock_db.go
 github.com/ava-labs/avalanchego/x/sync=Client=x/sync/mock_client.go
+github.com/ava-labs/avalanchego/database=Database=database/mock_database.go
```
Comment (evlekht) [inline scripts/mocks.mockgen.txt:51]: This is our mocks, that generator can process.

File: scripts/mocks.mockgen.txt [hunk 4]
```diff
@@ -12,16 +12,13 @@ github.com/ava-labs/avalanchego/snow/engine/avalanche/vertex=LinearizableVM=snow
 github.com/ava-labs/avalanchego/snow/engine/snowman/block=BuildBlockWithContextChainVM=snow/engine/snowman/block/mocks/build_block_with_context_vm.go
 github.com/ava-labs/avalanchego/snow/engine/snowman/block=ChainVM=snow/engine/snowman/block/mocks/chain_vm.go
 github.com/ava-labs/avalanchego/snow/engine/snowman/block=StateSyncableVM=snow/engine/snowman/block/mocks/state_syncable_vm.go
-github.com/ava-labs/avalanchego/snow/engine/snowman=Engine=snow/engine/snowman/mock_engine.go
```
Comment (evlekht) [inline scripts/mocks.mockgen.txt:14]: All this changes are just reverting diff that we have with avax. Probably was lost in previous merges.

File: staking/tls.go [hunk 5]
```diff
@@ -176,15 +172,3 @@ func NewCertAndKeyBytesWithSecpKey(secpKey *secp256k1.PrivateKey) ([]byte, []byt
 	}
 	return certBuff.Bytes(), keyBuff.Bytes(), nil
 }
-
-func VerifyCertificate(cert *x509.Certificate) error {
```
Comment (evlekht) [inline staking/tls.go:180]: All this changes are just reverting diff that we have with avax. Probably was lost in previous merges.

File: vms/proposervm/block/block.go [hunk 6]
```diff
@@ -102,18 +103,15 @@ func (b *statelessBlock) initialize(bytes []byte) error {
 		return nil
 	}
 
-	cert, err := staking.ParseCertificate(b.StatelessBlock.Certificate)
+	tlsCert, err := x509.ParseCertificate(b.StatelessBlock.Certificate)
 	if err != nil {
 		return fmt.Errorf("%w: %s", errInvalidCertificate, err)
 	}
 
-	if err := staking.VerifyCertificate(cert); err != nil {
```
Comment (evlekht) [inline vms/proposervm/block/block.go:111]: Just reverting diff that we have with avax. Probably was lost in previous merges.

File: network/peer/upgrader.go [hunk 7]
```diff
@@ -110,3 +110,11 @@ func CertToID(cert *x509.Certificate) (ids.NodeID, error) {
 	}
 	return ids.ToNodeID(pubKeyBytes)
 }
+
+func StakingCertToID(cert *staking.Certificate) (ids.NodeID, error) {
```
Comment (evlekht) [inline network/peer/upgrader.go:114]: Just a helper func, there is case, where we only have `staking.Certificate`.

File: scripts/camino_mocks.mockgen.txt [hunk 8]
```diff
@@ -0,0 +1,16 @@
+// TODO @evlekht
+// add this to 'scripts/mocks.mockgen.txt' when mockgen will be able to process this files correctly (some generics issues)
+
+github.com/ava-labs/avalanchego/cache=Cacher=cache/mock_cacher.go
```
Comment (evlekht) [inline scripts/camino_mocks.mockgen.txt:4]: Those mocks are created manually. Even latest mockgen (0.4.0) is failing on those interfaces.

File: ids/node_id.go [hunk 9]
```diff
@@ -76,12 +74,6 @@ func ToNodeID(bytes []byte) (NodeID, error) {
 	return NodeID(nodeID), err
 }
 
-func NodeIDFromCert(cert *staking.Certificate) NodeID {
```
Comment (evlekht) [inline ids/node_id.go]: We have import cycle conflict: `ids -> staking -> secp256k1fx -> ids`
I considered that moving `NodeIDFromCert` out of this package is the least diff and package-structure wise most reasonable change.
Its currently moved to `vms/proposervm` as its the only package that is uses it.

**Review by evlekht — COMMENTED** [review]
Not a review, just a comments for post-merge fixes.

**Review by knikos — APPROVED** [review]
lgtm. good job

**Pull request merged**